### DISCUSSION TITLE
ZSH-like partial match history

### DISCUSSION THREAD

**Discussion by harryeffinpotter** [chrisant996/clink#599] [General]
Hey I was wondering is there any way to get a zsh-liek history? What I mean is if I type ssh and then press the up arrow multiple times on zsh, it will scroll through every command I sent that started with ssh. Clink instead will find the last ssh command i sent when i type ssh but as soon as i hit the up arrow it just goes back to the last command I sent, overriding the ssh pretext.

**Comment by chrisant996** [comment]
`history-search-backward`

Search for that in the docs, and the sections on customizing key bindings.

**Reply by harryeffinpotter** [reply]
Thanks man!

**Reply by harryeffinpotter** [reply]
Yo I finally got around to this, and I applied your custom config, and excuse my french but **HOLY FUCKING SHIT** youre blowing my mind dude, you are a god. you should make this shit for the goddamn linux command line, this is BETTER than zsh, seriously dude i tried your config as is and the up arrow not only works the way i want it, but i discovered while reading the docs that page up acts the way i wanted up arrow to act, so itried pageup with your config cuz i didnt know what this history thing is, and holy SHIT once again, this shit is amazing. i already tell everyone to install clink, but now i have to tell them to use your config that you have in the docs, my god dude, blown. the fuck. away. im donating money to you right now one way or another, thank you for real. you make cmd not suck ass.

**Reply by harryeffinpotter** [reply]
Actually 2 questions

1. How can I donate to you for doing this incredible work?
2. Is there any place where users can share their LUA scripts/configs/etc? I'd really love to participate in such a place and also get some configs/scripts from others? I could see it being an awesome place, kind of like Directory Opus's forums, if you're familiar with that software. This is on the level of DOPUS and you are just one man. Obviously it is open source and there are many contribs but this shit goes so deep dude holy smokes. Love it.

**Reply by chrisant996** [reply]
> How can I donate to you for doing this incredible work?

I appreciate the appreciation, but I don't accept donations at this time.

> Is there any place where users can share their LUA scripts/configs/etc?

#157 has thoughts on that.  There's also the [Scripts](https://github.com/chrisant996/clink/discussions/categories/scripts) category in the Discussions forum.
